quinn-proto/src/connection/mod.rs

```diff
@@ -1228,6 +1228,16 @@ impl Connection {
         self.path.remote
     }
 
+    /// The remote handshake cid
```

Comment:
Are you sure this is the one you want? I would've guess you might need `initial_dst_cid`?

Reply:
The initial dst cid is set to a random 20 bytes by quinn client upon connecting, so it doesn't correlate to anything a custom ConnectionIdGenerator may produce